h/static/scripts/directive/excerpt.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+/**
+ * @ngdoc directive
+ * @name excerpt
+ * @restrict E
+ * @description This directive truncates its contents to a height specified in
+ *              CSS, and provides controls for expanding and collapsing the
+ *              resulting truncated element. For example, with the following
+ *              template HTML:
+ *
+ *                  <article class="post">
+ *                    <excerpt>
+ *                      <div class="body" ng-model="post.body"></div>
+ *                    </excerpt>
+ *                  </article>
+ *
+ *              You would need to define the allowable height of the excerpt in
+ *              CSS:
+ *
+ *                  article.post .excerpt {
+ *                    max-height: 10em;
+ *                  }
+ *
+ *              And the excerpt directive will take care of the rest.
+ *
+ *              You can selectively disable truncation by providing a boolean
+ *              expression to the `enabled` parameter, e.g.:
+ *
+ *                  <excerpt enabled="!post.inFull">...</excerpt>
+ */
+module.exports = function () {
+  return {
+    link: function (scope, elem, attr, ctrl) {
```

Comment:
For consistency with other directives I think it would make sense to expose properties on a named object rather than directly on the scope. That can either be done with a controller or just a named object on $scope